.gitmodules

```diff
@@ -31,3 +31,6 @@
 [submodule "arch/cpu/nrf52840/lib/nrf52-sdk"]
 	path = arch/cpu/nrf52840/lib/nrf52-sdk
 	url = https://github.com/contiki-ng/nrf-sdk-contiki-ng.git
+[submodule "arch/cpu/arm/CMSIS"]
+	path = arch/cpu/arm/CMSIS
```

Comment:
We could put the submodule under `arch/cpu/arm`? And submodule initialisation would create a `CSMSIS` sub-dir. Right?

Reply:
But the submodule will always be under a CMSIS dir, right? So this structure will create `arch/cpu/arm/CMSIS/CMSIS`. What have I missed?

Reply:
That is correct. So at the moment submodule is under CMSIS (arch/cpu/arm/CMSIS).
However the submodule also contains another folder called CMSIS in which the files are located.

ARM root folder:
```console
rosario@yagos-macbook-pro arm % pwd
/Users/rosario/contiki-ng/arch/cpu/arm
rosario@yagos-macbook-pro arm % ls -lah
total 40
drwxr-xr-x  10 rosario  staff   320B Feb  9 18:32 .
drwxr-xr-x  11 rosario  staff   352B Feb  9 18:32 ..
-rw-r--r--@  1 rosario  staff   6.0K Feb  9 18:32 .DS_Store
drwxr-xr-x  16 rosario  staff   512B Feb  9 18:32 CMSIS
-rw-r--r--   1 rosario  staff   2.3K Feb  9 18:30 Makefile.arm
-rw-r--r--   1 rosario  staff   3.2K Feb  9 18:30 arm-def.h
drwxr-xr-x   4 rosario  staff   128B Feb  9 18:30 common
drwxr-xr-x   8 rosario  staff   256B Feb  9 18:30 cortex-m
-rw-r--r--   1 rosario  staff    62B Feb  9 18:30 doxygen-group.txt
drwxr-xr-x   4 rosario  staff   128B Feb  9 18:30 openocd
rosario@yagos-macbook-pro arm % 

```

submodule root folder
```console
rosario@yagos-macbook-pro CMSIS % pwd
/Users/rosario/contiki-ng/arch/cpu/arm/CMSIS
rosario@yagos-macbook-pro CMSIS % ls -lah
total 3920
drwxr-xr-x  16 rosario  staff   512B Feb  9 18:32 .
drwxr-xr-x  10 rosario  staff   320B Feb  9 18:32 ..
-rw-r--r--@  1 rosario  staff   6.0K Feb  9 18:32 .DS_Store
-rw-r--r--   1 rosario  staff    52B Feb  9 18:32 .git
-rw-r--r--   1 rosario  staff   676B Feb  9 18:32 .gitattributes
-rw-r--r--   1 rosario  staff   224B Feb  9 18:32 .gitignore
-rw-r--r--   1 rosario  staff   263K Feb  9 18:32 ARM.CMSIS.pdsc
drwxr-xr-x  15 rosario  staff   480B Feb  9 18:32 CMSIS
-rw-r--r--   1 rosario  staff   1.2M Feb  9 18:32 CMSIS_Review_Meeting_2020.pdf
drwxr-xr-x   5 rosario  staff   160B Feb  9 18:32 Device
-rw-r--r--   1 rosario  staff    11K Feb  9 18:32 LICENSE.txt
-rw-r--r--   1 rosario  staff    11K Feb  9 18:32 README.md
-rw-r--r--   1 rosario  staff   431K Feb  9 18:32 Scalable Software Stack.pdf
drwxr-xr-x  16 rosario  staff   512B Feb  9 18:32 docs
-rw-r--r--   1 rosario  staff    12K Feb  9 18:32 linter.py
-rw-r--r--   1 rosario  staff   2.1K Feb  9 18:32 manifest
rosario@yagos-macbook-pro CMSIS % 

```

Reply:
Duh. I'm with you now.